contracts/strategies/tetu-v1/TetuV1SingleTokenStrictStrategy.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@tetu_io/tetu-contracts-v2/contracts/strategy/StrategyStrictBase.sol";
+import "@tetu_io/tetu-contracts-v2/contracts/interfaces/ITetuLiquidator.sol";
+import "../../integrations/tetu-v1/ISmartVault.sol";
+import "../../tools/AppLib.sol";
+import "../../tools/ERC20Helpers.sol";
+
+/// @title Simple auto compounding strategy for TETU V1 vaults.
+/// @author AlehNat
+contract TetuV1SingleTokenStrictStrategy is StrategyStrictBase, ERC20Helpers {
+  using SafeERC20 for IERC20;
+
+  string public constant override NAME = "TetuV1 Single Token Strict Strategy";
+  string public constant override PLATFORM = "TETU";
+  string public constant override STRATEGY_VERSION = "1.0.0";
+
+  uint private constant _ASSET_LIQUIDATION_SLIPPAGE = 500; // 0.5%
+
+  // in this strategy TETU V1 vault is used as a pool
+  ISmartVault public immutable pool;
+  ITetuLiquidator public immutable liquidator;
+  address public immutable xTetuAddress;
+
+  bool public override isReadyToHardWork;
+
+  constructor(address _pool, address _liquidator, address _xTetuAddress) {
+    pool = ISmartVault(_pool);
+    liquidator = ITetuLiquidator(_liquidator);
+    xTetuAddress = _xTetuAddress;
+    isReadyToHardWork = true;
+  }
+
+
+  // uint earned, uint lost is it in USD?
+  function doHardWork() external override returns (uint earned, uint lost) {
+    uint assetBalanceBefore = _balance(asset);
+    _claim();
+    _unwrapXTetu();
+    _liquidateReward();
+    uint assetBalance = _balance(asset);
+    if (assetBalance > 0) {
+      _depositToPool(assetBalance);
+    }
+    earned = 0;
+    lost = 0;
+    if(assetBalance > assetBalanceBefore) {
```

Comment:
do not think it will reflect earned amount

Reply:
need to check the whole balance is under control of this strategy before and after

Reply:
fixed